packages/framework/esm-utils/src/patient-helpers.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Gets the display name for a patient.
+ * @param fhirPatient The patient details in FHIR format.
+ * @returns The patient's display name.
+ */
+export function displayName(fhirPatient): string {
+  return fhirPatient.name?.[0]?.text ?? fallbackDisplayName(fhirPatient);
+}
+
+function fallbackDisplayName(fhirPatient): string {
+  return `${fhirPatient.name?.[0].given?.join(' ')} ${fhirPatient?.name?.[0].family}`;
```

Comment:
For a library, it's best to get this right, i.e., never return "undefined" or "undefined undefined":

```suggestion
   if (!fhirPatient.name?.[0]) {
     return '';
   }
   
   const name = `${fhirPatient.name[0].given?.join(' ') ?? ''}`;
   
   return fhirPatient.name[0].family ? (name.length > 0 ? ' ' : '') + `fhirPatient.name[0].family` : name;
```

Reply:
OK. I have actually rewritten this code substantially to introduce functionality to `selectPreferredName`...
I have ensured I never return "undefined"